src/routes/v0/stream/queries.rs

```diff
@@ -0,0 +1,195 @@
+use crate::routes::v0::queries::PaginationQuery;
+use serde::de::{self, Deserializer, MapAccess, Visitor};
+use serde::{Deserialize, Serialize};
+use std::fmt;
+use utoipa::ToSchema;
+
+#[derive(Debug, Serialize, Deserialize, ToSchema, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum PostStreamSorting {
+    Timeline,
+    TotalEngagement,
+}
+
+#[derive(Deserialize, ToSchema, Debug, Clone, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum ViewerStreamSource {
```

Comment:
Do we need both `ViewerStreamSource` and `ViewerStreamSourceQuery` ?

I guess, now that we have broken apart `viewer_id` into the two usages: 1. classical `viewer_id` use for relationships and new 2. `observer_id` for post stream curation, we can rename these structs as `StreamSource` and `StreamSourceQuery`